networking_ccloud/extensions/fabricoperations.py

```diff
@@ -139,3 +146,105 @@ def index(self, request, **kwargs):
 
         return resp
+
+
+class SyncController(wsgi.Controller):
+    def __init__(self, db):
+        super().__init__()
+        self.db = db
+        self.drv_conf = get_driver_config()
+        self.plugin = directory.get_plugin()
+
+    @check_cloud_admin
+    def index(self, request, **kwargs):
+        return {"driver_reached": True}
+
+    @check_cloud_admin
+    def show(self, request, **kwargs):
+        obj_type = kwargs.pop("id")
+        # FIXME: Is this validation something we could do via an API scheme?
+        if obj_type not in ("switches", "networks"):
+            raise web_exc.HTTPBadRequest("Invalid sync option, please choose either switches or networks")
+
+        objs = request.params.getall('obj')
+        if not isinstance(objs, list) or not all(isinstance(e, str) for e in objs):
+            raise web_exc.HTTPBadRequest("Payload must be a list of strings")
+
+        if not objs:
+            raise web_exc.HTTPBadRequest(f"Please provide at least one object identifier for the {obj_type} "
+                                         f"you want to work with")
+
+        # FIXME: command via GET is nothing we should do in the end product, fix this to proper API something
+        # FIXME: write proper tests for this sync logic
+        cmd = request.params.get('cmd', 'config')
+        if cmd not in ('config', 'diff', 'sync'):
+            raise web_exc.HTTPBadRequest(f"Cmd '{cmd}' is invalid, choose from config, diff, sync")
+
+        filter_switches = None
+        LOG.error("Test ctx %s sess %s", request.context, getattr(request.context, "session", None))
+        if obj_type == "networks":
+            # validate that all networks exist
+            for network_id in objs:
+                try:
+                    self.plugin.get_network(request.context, network_id)
+                except nl_exc.NetworkNotFound as e:
+                    raise web_exc.HTTPBadRequest(str(e))
+
+            # query db for physnets'n'stuff
+            # create config update list for each bindinghost of each network
+            scul = agent_msg.SwitchConfigUpdateList(agent_msg.OperationEnum.add, self.drv_conf)
+            net_segments = self.db.get_hosts_on_segments(request.context, network_ids=objs)
+        elif obj_type == "switches":
+            # make sure switches exist and get their attached physnets
+            physnets = []
+            for switch in objs:
+                if not self.drv_conf.get_switch_by_name(switch):
+                    raise web_exc.HTTPBadRequest(f"Switch '{switch}' does not exist in current driver's switch config")
+                # FIXME: can we have switches without a switchgroup?
+                sg = self.drv_conf.get_switchgroup_by_switch_name(switch)
+                physnets.append(sg.vlan_pool)
+
+            filter_switches = objs
+            # query db
+            net_segments = self.db.get_hosts_on_segments(request.context, physical_networks=physnets)
```

Comment:
Yes, this could be an empty list, e.g. for a network that is present in OpenStack, but has no bindings that belong to the fabric (just other network parts, e.g. ACI). Reading the sqlalchemy1.2 changelog reads though as this is not a problem and will result in a `1 != 1` statement being rendered, see here: https://github.com/zzzeek/sqlalchemy/commit/f3b6f4f8da5223fae0a1dd948d4266b2e49e317c

Reply:
Ah.  I need to remember that. Thank you.